src/cudamatrix/cu-device.h

```diff
@@ -55,7 +55,12 @@ class CuDevice {
   inline void* Malloc(size_t size) { return allocator_.Malloc(size); }
 
   inline void* MallocPitch(size_t row_bytes, size_t num_rows, size_t *pitch) {
-    return allocator_.MallocPitch(row_bytes, num_rows, pitch);
+    if (random_stride_mode_) {
+      return allocator_.MallocPitch(row_bytes + 128 * (rand() & 1), num_rows,
```

Comment:
The stride bucket size is hardware dependent. It seems it is keep increasing. On K40c with CUDA 7.5 it is 512.
http://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#device-memory-accesses